### PR TITLE
deps(dev): upgrade aegir to `38.1.2`

### PR DESCRIPTION
**Pull request by maschad**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -186,16 +186,14 @@
...
     "protons-runtime": "^4.0.1",
+    "uint8arraylist": "^2.4.3",
     "uint8arrays": "^4.0.2"
   },
   "devDependencies": {
     "@types/mocha": "^10.0.0",
-    "aegir": "^37.0.12",
+    "aegir": "^38.1.2",
     "benchmark": "^2.1.4",
-    "protons": "^6.0.0",
```
Comment (achingbrain) [inline package.json:195]: `protons` is used in the `generate` npm script.

Please update the `dep-check` command to ignore protons - `"dep-check": "aegir dep-check -i protons"`

File: src/hmac/index.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import crypto from 'crypto'
 import lengths from './lengths.js'
 
-export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array) { // eslint-disable-line require-await
+export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<{ digest: (data: Uint8Array) => Promise<Buffer>, length: number }> {
```
Comment (achingbrain) [inline src/hmac/index.ts:4]: When you have a type like this with multiple fields it's usually better to export an interface and use that in the return type:

```suggestion
export interface Hmac {
  digest: (data: Uint8Array) => Promise<Uint8Array>
  length: number
}

export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<Hmac> {
```

File: src/keys/ed25519-class.ts [hunk 3]
```diff
@@ -51,30 +52,30 @@ export class Ed25519PrivateKey {
     this._publicKey = ensureKey(publicKey, crypto.publicKeyLength)
   }
 
-  async sign (message: Uint8Array) { // eslint-disable-line require-await
+  async sign (message: Uint8Array): Promise<Buffer> { // eslint-disable-line require-await
```
Comment (achingbrain) [inline src/keys/ed25519-class.ts:55]:
```suggestion
  async sign (message: Uint8Array): Promise<Uint8Array> { // eslint-disable-line require-await
```

File: src/keys/ed25519.ts [hunk 4]
```diff
@@ -2,6 +2,7 @@ import crypto from 'crypto'
 import { promisify } from 'util'
 import { toString as uint8arrayToString } from 'uint8arrays/to-string'
 import { fromString as uint8arrayFromString } from 'uint8arrays/from-string'
+import type { Uint8ArrayKeyPair } from './interface'
```
Comment (achingbrain) [inline src/keys/ed25519.ts:5]:
```suggestion
import type { Uint8ArrayKeyPair } from './interface.js'
```

File: src/keys/jwk2pem.ts [hunk 5]
```diff
@@ -3,14 +3,14 @@ import 'node-forge/lib/rsa.js'
 import forge from 'node-forge/lib/forge.js'
 import { base64urlToBigInteger } from '../util.js'
 
-function convert (key: any, types: string[]) {
+function convert (key: any, types: string[]): any {
```
Comment (achingbrain) [inline src/keys/jwk2pem.ts:6]: Is it possible to lose the `any`?  I think the type should be `forge.jsbn.BigInteger[]`?

File: src/keys/jwk2pem.ts [hunk 6]
```diff
@@ -3,14 +3,14 @@ import 'node-forge/lib/rsa.js'
...
   return types.map(t => base64urlToBigInteger(key[t]))
 }
 
-export function jwk2priv (key: JsonWebKey) {
+export function jwk2priv (key: JsonWebKey): { encrypt: (msg: string) => string, decrypt: (msg: string) => string } {
```
Comment (achingbrain) [inline src/keys/jwk2pem.ts:10]: Please convert to an interface

File: src/keys/jwk2pem.ts [hunk 7]
```diff
@@ -3,14 +3,14 @@ import 'node-forge/lib/rsa.js'
...
   return forge.pki.setRsaPrivateKey(...convert(key, ['n', 'e', 'd', 'p', 'q', 'dp', 'dq', 'qi']))
 }
 
-export function jwk2pub (key: JsonWebKey) {
+export function jwk2pub (key: JsonWebKey): { encrypt: (msg: string) => string, decrypt: (msg: string) => string } {
```
Comment (achingbrain) [inline src/keys/jwk2pem.ts:14]: Please convert to an interface

File: src/keys/rsa-class.ts [hunk 8]
```diff
@@ -56,42 +57,42 @@ export class RsaPrivateKey {
     this._publicKey = publicKey
   }
 
-  genSecret () {
+  genSecret (): Uint8Array {
     return crypto.getRandomValues(16)
   }
 
-  async sign (message: Uint8Array) { // eslint-disable-line require-await
+  async sign (message: Uint8Array): Promise<Buffer> { // eslint-disable-line require-await
```
Comment (achingbrain) [inline src/keys/rsa-class.ts:64]:
```suggestion
  async sign (message: Uint8Array): Promise<Uint8Array> { // eslint-disable-line require-await
```

File: src/keys/rsa.ts [hunk 9]
```diff
@@ -42,14 +42,14 @@ export async function unmarshalPrivateKey (key: JsonWebKey): Promise<JWKKeyPair>
 
 export { randomBytes as getRandomValues }
 
-export async function hashAndSign (key: JsonWebKey, msg: Uint8Array) { // eslint-disable-line require-await
+export async function hashAndSign (key: JsonWebKey, msg: Uint8Array): Promise<Buffer> { // eslint-disable-line require-await
```
Comment (achingbrain) [inline src/keys/rsa.ts:45]:
```suggestion
export async function hashAndSign (key: JsonWebKey, msg: Uint8Array): Promise<Uint8Array> { // eslint-disable-line require-await
```

File: src/keys/rsa.ts [hunk 10]
```diff
@@ -58,12 +58,12 @@ export async function hashAndVerify (key: JsonWebKey, sig: Uint8Array, msg: Uint
 
 const padding = crypto.constants.RSA_PKCS1_PADDING
 
-export function encrypt (key: JsonWebKey, bytes: Uint8Array) {
+export function encrypt (key: JsonWebKey, bytes: Uint8Array): Buffer {
```
Comment (achingbrain) [inline src/keys/rsa.ts:61]:
```suggestion
export function encrypt (key: JsonWebKey, bytes: Uint8Array): Uint8Array {
```

File: src/keys/rsa.ts [hunk 11]
```diff
@@ -58,12 +58,12 @@ export async function hashAndVerify (key: JsonWebKey, sig: Uint8Array, msg: Uint
...
   // @ts-expect-error node types are missing jwk as a format
   return crypto.publicEncrypt({ format: 'jwk', key, padding }, bytes)
 }
 
-export function decrypt (key: JsonWebKey, bytes: Uint8Array) {
+export function decrypt (key: JsonWebKey, bytes: Uint8Array): Buffer {
```
Comment (achingbrain) [inline src/keys/rsa.ts:66]:
```suggestion
export function decrypt (key: JsonWebKey, bytes: Uint8Array): Uint8Array {
```

File: src/aes/ciphers-browser.ts [hunk 12]
```diff
@@ -5,7 +5,7 @@ import forge from 'node-forge/lib/forge.js'
 import { toString as uint8ArrayToString } from 'uint8arrays/to-string'
 import { fromString as uint8ArrayFromString } from 'uint8arrays/from-string'
 
-export function createCipheriv (mode: any, key: Uint8Array, iv: Uint8Array) {
+export function createCipheriv (mode: any, key: Uint8Array, iv: Uint8Array): { update: (data: Uint8Array) => Uint8Array } {
```
Comment (achingbrain) [inline src/aes/ciphers-browser.ts:8]: Please convert to an interface

File: src/aes/ciphers-browser.ts [hunk 13]
```diff
@@ -16,7 +16,7 @@ export function createCipheriv (mode: any, key: Uint8Array, iv: Uint8Array) {
   }
 }
 
-export function createDecipheriv (mode: any, key: Uint8Array, iv: Uint8Array) {
+export function createDecipheriv (mode: any, key: Uint8Array, iv: Uint8Array): { update: (data: Uint8Array) => Uint8Array } {
```
Comment (achingbrain) [inline src/aes/ciphers-browser.ts:19]: Please convert to an interface

**Review by achingbrain — CHANGES_REQUESTED** [review]
Nearly there, just a few simple changes to make, comments inline.

File: package.json (hunk 1 above)
Comment (maschad) [reply package.json:195]: Are there any other packages that `dep-check` might ignore? May be worthwhile to revisit the functionality of `dep-check` if this isn't the only exception.

Comment (achingbrain) [reply package.json:195]: It only tests for modules used in code, it doesn't include npm scripts.

I suppose you could get aegir to look at the `bin` object of every dependency and see if any of the commands are used in the npm scripts of the current project and add the containing module to the ignore list if so but it's a lot of work and may end up with some false positives if two deps export the same command.

Probably not worth the effort IMO.

File: src/hmac/index.ts [hunk 14]
```diff
@@ -1,7 +1,12 @@
 import crypto from 'crypto'
 import lengths from './lengths.js'
 
-export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array) { // eslint-disable-line require-await
+export interface Hmac {
+  digest: (data: Uint8Array) => Promise<Uint8Array>
+  length: number
+}
+
+export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<Hmac> {
```
Comment (achingbrain) [inline src/hmac/index.ts:9]: HMAC is an acronym, we capitalise acronyms elsewhere in the codebase so rename for consistency:

```suggestion
export interface HMAC {
  digest: (data: Uint8Array) => Promise<Uint8Array>
  length: number
}

export async function create (hash: 'SHA1' | 'SHA256' | 'SHA512', secret: Uint8Array): Promise<HMAC> {
```

**Review by achingbrain — APPROVED** [review]

**Pull request merged**